x/feegrant/keeper/keeper.go

```diff
@@ -227,20 +227,19 @@ func (k Keeper) UseGrantedFees(ctx context.Context, granter, grantee sdk.AccAddr
 	}
 
 	remove, err := grant.Accept(context.WithValue(ctx, corecontext.EnvironmentContextKey, k.Environment), fee, msgs)
-	if remove && err == nil {
-		// Ignoring the `revokeFeeAllowance` error, because the user has enough grants to perform this transaction.
-		_ = k.revokeAllowance(ctx, granter, grantee)
-
-		return k.emitUseGrantEvent(ctx, granterStr, granteeStr)
-	}
 	if err != nil {
 		return err
 	}
+
 	if err := k.emitUseGrantEvent(ctx, granterStr, granteeStr); err != nil {
 		return err
 	}
 
-	// if fee allowance is accepted, store the updated state of the allowance
+	if remove {
```

Comment:
That's a behavior change, the comment above said it was ignoring the revokeAllowance error on remove. This will return it, which doesn't match the comment above.